File: Search_Insert_Position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
        int left = 0;
        int right = nums.length-1;
        int ans = -1;
        
        while ( left < right ){
            
            int mid = left + ( right - left ) / 2;
            
            if ( target == nums[mid] ){
                ans = mid;
                return mid;
            }
            else if ( target > nums[mid] ){
                left = mid+1;
            }
            else {
                right = mid-1;
            }
            
        }
        if ( ans == -1 ) {
			if ( target > nums[left] ) {
				ans = left+1;
			}
			else {
				ans = left;
			}
		}
        return ans;
    }
}

File: Coding Problems & My Solutions/Search Insert Position/Search_Insert_Position.swift
class Solution {
    func searchInsert(_ nums: [Int], _ target: Int) -> Int {
        var l = 0
        var r = nums.count - 1
        while l <= r {
            let m = (l + r) / 2
            if nums[m] == target {
                return m
            }
            else if target > nums[m] {
                l = m + 1
            }
            else {
                r = m - 1
            }
        }
        return l
    }
}